staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -544,13 +544,8 @@ func CanonicalVerb(verb string, scope string) string {
 // LIST and APPLY from PATCH.
 func CleanVerb(verb string, request *http.Request) string {
 	reportedVerb := verb
-	if verb == "LIST" {
-		// see apimachinery/pkg/runtime/conversion.go Convert_Slice_string_To_bool
-		if values := request.URL.Query()["watch"]; len(values) > 0 {
-			if value := strings.ToLower(values[0]); value != "0" && value != "false" {
-				reportedVerb = "WATCH"
-			}
-		}
+	if verb == "LIST" && checkIfWatch(request) {
```

Comment:
Can you please also add a test:
GET isn't be transformed to WATCH if we have the right query param on the request

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics_test.go#L49

Reply:
done!
GET will be transformed to WATCH from 'getVerbIfWatch'  only in blew function
- RecordRequestAbort
- RecordRequestTermination
- RecordLongRunning